wgpu/src/api/device.rs

```diff
@@ -695,3 +695,36 @@ impl fmt::Display for Error {
         }
     }
 }
+
+// Copied from [`futures::task::noop_waker`].
+// Needed until MSRV is 1.85 with `task::Waker::noop()` available
+#[cfg(feature = "noop")]
+mod waker {
+    use core::ptr::null;
+    use core::task::{RawWaker, RawWakerVTable, Waker};
+
+    unsafe fn noop_clone(_data: *const ()) -> RawWaker {
+        noop_raw_waker()
+    }
+
+    unsafe fn noop(_data: *const ()) {}
+
+    const NOOP_WAKER_VTABLE: RawWakerVTable = RawWakerVTable::new(noop_clone, noop, noop, noop);
+
+    const fn noop_raw_waker() -> RawWaker {
+        RawWaker::new(null(), &NOOP_WAKER_VTABLE)
+    }
+
+    /// Get a static reference to a [`Waker`] which
+    /// does nothing when `wake()` is called on it.
+    #[inline]
+    pub fn noop_waker_ref() -> &'static Waker {
+        struct SyncRawWaker(RawWaker);
+        unsafe impl Sync for SyncRawWaker {}
+
+        static NOOP_WAKER_INSTANCE: SyncRawWaker = SyncRawWaker(noop_raw_waker());
+
+        // SAFETY: `Waker` is #[repr(transparent)] over its `RawWaker`.
+        unsafe { &*(&NOOP_WAKER_INSTANCE.0 as *const RawWaker as *const Waker) }
+    }
+}
```

Comment:
Instead of vendoring this, could we use the `noop_waker` crate?

Reply:
Yep! That's what I originally did. So the crate hasn't been updated in a while and I didn't recognize the creator. And then I saw futures had a different implementation, and there must be a reason why. I figured this crate might already rely on futures as other ones do and that would solve it. But it does not depend on futures sadly.

So I looked at the underlying implementations, and they were very small. Because this code is gated to a feature, won't be developed, and will be ripped out later I felt the small amount was better than an added dependency.

Happy to switch it to the dep if you prefer.

Reply:
Nice explanation, I agree!